src/api/page.js

```diff
@@ -32,18 +38,26 @@ function hasTag(aPage, aTag) {
     return false;
 }
 
-// Optional path, defaults to current page
-//
-// Optional depth. Number of levels of children to include, 0
-// is the path page
-//
-// Optional self, defaults to false. Include the path page in
-// the results
-//
-// This is not called by any macros, and is only used here by
-// wiki.tree(), so we could move it to be part of that function.
+/**
+ * Optional path, defaults to current page
+ *
+ * Optional depth. Number of levels of children to include, 0
+ * is the path page
+ *
+ * Optional self, defaults to false. Include the path page in
+ * the results
+ *
+ * This is not called by any macros, and is only used here by
+ * wiki.tree(), so we could move it to be part of that function.
+ *
+ * @param {string} [path]
+ * @param {number|string} [depth]
+ * @param {number|boolean} [self]
+ * @return {Promise<any[]>}
+ */
 async function subpages(path, depth, self) {
     var url = util.apiURL((path ? path : this.env.url) + '$children');
+    // @ts-ignore: Unsafe `parseInt(number)` call
     var depth_check = parseInt(depth);
```

Comment:
This is unsafe because `0.0000001` stringifies to `1e-7`, which results in `parseInt(0.0000001)` returning `1`.

Same for `1000000000000000000000` (stringifies to `1e+21`).